### DISCUSSION TITLE
Allow setting a seed/key for temporary passwords

### DISCUSSION THREAD

**Discussion by qgustavor** [rustdesk/rustdesk#5412] [Feature Request]
Currently there is a one-time password option with a single option that changes its length, but there is no option to set a seed/key. At the moment [the code that handles that](https://github.com/rustdesk/rustdesk/blob/542d86b667a2da19f89591a03a39edd9ab4d406e/libs/hbb_common/src/config.rs#L757-L762) just generates a random password based on this single option. If there was a way to set a seed/key using [TOTP](https://en.wikipedia.org/wiki/Time-based_one-time_password) then it users could use a myriad of applications like Google Authenticator, Microsoft Authenticator, Aegis [and many others](https://en.wikipedia.org/wiki/Comparison_of_OTP_applications) to access temporary passwords in a safe manner. [Like GitHub](https://github.blog/changelog/2023-06-29-totp-codes-for-2fa-are-now-single-use/) some mechanism should be included to detect and block password reuse. Also it should detect and block password brute forcing. Likewise, if users find it insecure to use just 6 digits as passwords (which is a fairly common setup) most applications allows the generation of codes with more than 6 digits.

An addendum: I don't know well why the current temporary password mechanism is made for. If the password is generated randomly locally, does it mean that's just a password meant to be transmitted over a third party to give one-time access to a computer by someone with local access to this computer? I use RustDesk to remotely connect to my desktop computer, I liked the one-time password idea because I thought it would allow me if I'm in a hurry to borrow a computer from someone that allows me to run RustDesk on it and access my computer and files - I can access them on my phone, but borrowing a computer is better because phone screens are small. As an example, the library from college have RustDesk installed already. I would not need to worry about a keylogger getting my permanent password, I could just get the temporary password on my phone and type on their computer. Then I guess the use case for this feature is more to aid people asking for support.

**Comment by xlionjuan** [comment]
In simple English.....you want to use TOTP to be your temporary password?

**Reply by xlionjuan** [reply]
If true, this sounds like a good idea, but I didn't heard any remote desktop software implemented this.

**Reply by qgustavor** [reply]
Yeah. Well, to be fair, I'm totally fine if you don't want to implement this, I know that OSS development is hard.

**Reply by michael-rzn** [reply]
Personally i would love the option of signing using only a TOPT and avoididng the whole fixed pasword saga. I agree that it seems most password flows are designed with a 3rd party's convenience and only a modest interest in ensuring that Im the only one that cna access my account and that if i forget or lose one piece of the puzzle i can still easily get back in. EMailing codes being the main culprit there of course, so any thing that prevents emailing of recovery or access codes in its entireity im sure would make things a lot more secure for many people who struggle to achieve email security. :)
